### PR TITLE
Added crawler for 4chan.org

### PR DESCRIPTION
**Pull request by pathim**
wow 
such python
very beer
amaze
wooooow
kthxbye

### CODE REVIEW

**Comment by jkowalleck** [comment]
Indent seams to be inconsistent after first review. This may cause python print warnings or maybe just prevent it from running.  Can someone merge manually and check run in local repo,  before the pullreq is accepted? 

Code quality looks good, since its basically a copy of one of the existing crawlers with some small changes.

File: crawler/fourchan.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    def _restart_at_front(self):
+        self.__next = ""
+
+    def __init__(self, uri):
+        self.__uri = self.__class__.__build_uri(uri)
+        self._restart_at_front()
+
+    def _crawl(self):
+	self.__class__._log("debug", "uri is: %s, next is: %s" % (self.__uri, self.__next))
```
Comment (jkowalleck) [inline crawler/fourchan.py]: Inconsistent indent?  Check and push fix,  plz

**Comment by noqqe** [comment]
@jkowalleck will do :)

**Comment by jkowalleck** [comment]
i am on a train ride - as usual, wehn i work in this project ;-) here is what i have done so far: 

----

how to process a pull request manually for integration test
    - or - 
how to check if a merge of a remote branch or pull request merge will break something

1 fresh clone parent repo: git clone <remote-repo>.git
2 add the foreign remote repo: git remote add <foreign-alias> <foreign-remotre-repo>.git
3 fetch foreign remote: git fetch <foreign-alias>
3 switch to the target branch for the merge: git checkout origin/<target-branch>
4 merge foreign source branch into current branch: git merge <foreign-alias>/<foreign-remote-branch>

how this looked in my example: 
~~~sh
git clone https://github.com/k4cg/nichtparasoup.git # origin = k4cg
cd nichtparasoup

git remote add fablab https://github.com/fablabnbg/nichtparasoup.git

# update k4cg sources and checkout it's master to local
git fetch origin
git checkout origin/master

# update fablab sources and merge foreign master into local (master)
git fetch fablab 
git merge fablab/master 

# status: foreign remote branch with incoming changes is merged into local version of target branch locally
~~~

so now you have a version, that represents the desired status of the pull request.

now run the program (and/or do your unit tests).

**Comment by jkowalleck** [comment]
@pathim  fixme: use 4 spaces as indent. I'll inline-comment on your changed files.

**Comment by noqqe** [comment]
@jkowalleck why ... uhm.. great howto, but you can see exactly this directly on github :)

**Comment by jkowalleck** [comment]
@noqqe sure ;-) but can github run program/unit tests and tell me what's wrong? :-P

File: crawler/fourchan.py (hunk 1 above)
Comment (jkowalleck) [inline crawler/fourchan.py]: @pathim  fixme: use 2 * (4  spaces) as indent. currently you used 1 tab

File: crawler/fourchan.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        response = urllib2.urlopen(request, timeout=self.__class__.timeout())
+
+        page = BeautifulSoup(response.read())
+
+        # get more content ("scroll down")
+        # to know what page to parse next
+        # update new last URI when we're not on first run
+        self.__next =page.find_all("a", {"class": "button"})[-1]["href"]
+
+	for con in page.find_all("a", {"class": "fileThumb"}):
```
Comment (jkowalleck) [inline crawler/fourchan.py]: @pathim  fixme: use 2 * (4  spaces) as indent. currently used 1 tab

**Comment by jkowalleck** [comment]
Hello @pathim 

thank you for your contribution. looks good to me.
could you change 2 minor indent issues and push the changes? I added annotations in the crawler file you submitted.
After that is done we will be happy to accept your pull request.

Best, 
Jan.

**Comment by noqqe** [comment]
@jkowalleck @pathim already did. just wait and relax a bit

**Comment by jkowalleck** [comment]
... with each new crawler there is a new source icon we need to add ...

@pathim, would you be so kind and tell which icon should represent the 4chan crawler? 
you may also contribute this icon yourself in the file `templates_raw/root/css/sourceIcons.css` if you like.
If you need any help with the template builder, feel free to ask. ;-)

Best,
Jan.

**Comment by noqqe** [comment]
merged, also updated readme

**Pull request merged**